### PR TITLE
Don't convert input bam files that are supplied in post-alignment steps, it's not necessary. 

### LINKED ISSUE #1162: Allow on restart to not convert the bam files to cram
### Description of feature

Dear nf-core community, thank you very much for your incredible work!

I'm working with WGS samples with 100 X coverage. I have bam and bai files.
When I run sarek on these files I have to wait from 4 to 6 hour to convert them from bam to cram and I don't even need those cram files.
I was wondering if it could be possible to create and option to work exclusively with bam instead of cram.

Furthermore, looking at the code and modules, I found that BAM_MARKDUPLICATES:GATK4_MARKDUPLICATES is the main gateway to cram instead of bam outputs.

https://github.com/nf-core/sarek/blob/3.1.2/modules/nf-core/gatk4/markduplicates/main.nf

Probably with a slight modification to this modules, making it optional to convert from bam to cram after markduplicates, it'll also be possible to have bam outputs without converting cram to bam when using the full pipeline starting from fastq inputs.

Thank you in advance and best regards!
Youssef

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
Thank you to @SPPearce for chatting with me again to finally remove this wasteful step.

Closes #1162

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: tests/.nftignore [hunk 1]
```diff
@@ -20,4 +21,6 @@ reports/fastqc/**/*_fastqc.{html,zip}
 reports/markduplicates/**/*.md.cram.metrics
 reports/snpeff/*/*_snpEff.csv
 reports/snpeff/*/snpEff_summary.html
+reports/vcftools/**/*.TsTv.qual
 variant_calling/**/*.vcf.{gz,gz.tbi}
+.DS_Store
```
Comment (maxulysse) [inline tests/.nftignore:26]: That should not be produce by Nextflow

File: tests/start_from_markduplicates.nf.test [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                    stable_name,
+                    // All files with stable contents
+                    stable_path,
+                    // All cram files
+                    cram_files.collect{ file -> [ file.getName(), cram(file.toString(), fasta).getReadsMD5() ] }
+                ).match() }
+            )
+        }
+    }
+
```
Comment (maxulysse) [inline tests/start_from_markduplicates.nf.test:162]:
```suggestion
```

File: tests/start_from_preparerecalibration.nf.test [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                    stable_name,
+                    // All files with stable contents
+                    stable_path,
+                    // All cram files
+                    cram_files.collect{ file -> [ file.getName(), cram(file.toString(), fasta).getReadsMD5() ] }
+                ).match() }
+            )
+        }
+    }
+
```
Comment (maxulysse) [inline tests/start_from_preparerecalibration.nf.test:162]:
```suggestion
```

File: tests/start_from_recalibration.nf.test [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                    stable_name,
+                    // All files with stable contents
+                    stable_path,
+                    // All cram files
+                    cram_files.collect{ file -> [ file.getName(), cram(file.toString(), fasta).getReadsMD5() ] }
+                ).match() }
+            )
+        }
+    }
+
```
Comment (maxulysse) [inline tests/start_from_recalibration.nf.test:162]:
```suggestion
```

**Review by maxulysse — DISMISSED** [review]

File: tests/.nftignore (hunk 1 above)
Comment (FriederikeHanssen) [reply tests/.nftignore:26]: it ended up in my nf-test test files repeatedly

**Review by maxulysse — APPROVED** [review]

**Pull request merged**